canister/src/blocktree.rs

```diff
@@ -88,6 +88,37 @@ impl BlockTree {
             children: vec![],
         }
     }
+
+    /// Returns all blocks in the tree with their depths
+    /// separated by heights.
+    pub fn blocks_with_depths_separated_by_heights(&self) -> Vec<Vec<(&Block, u32)>> {
+        let mut blocks_with_confirmations_by_height: Vec<Vec<(&Block, u32)>> = vec![vec![]];
```

Comment:
```suggestion
        let mut blocks_with_depths_by_height: Vec<Vec<(&Block, u32)>> = vec![vec![]];
```

Let's stick with depth rather that confirmations. I agree it's not exactly the same as the typical definition of depth in a CS tree, but bitcoin uses different definitions of height/depth anyways, so we can be consistent with those definitions.